pytorch_lightning/core/lightning.py

```diff
@@ -1146,8 +1174,77 @@ def backward(self, loss, optimizer, optimizer_idx):
                 loss.backward()
 
         """
-        if self.trainer.train_loop.automatic_optimization or self._running_manual_backward:
-            loss.backward(*args, **kwargs)
+        loss.backward(*args, **kwargs)
+
+    def _backward_wrapper(self, loss: Tensor, optimizer: Optimizer, optimizer_idx: int, *args, **kwargs) -> None:
+        automatic_optimization = self.trainer.train_loop.automatic_optimization
+
+        if automatic_optimization or self._running_manual_backward:
+
+            is_ddp = isinstance(self.trainer.model, LightningDistributedDataParallel)
+            never_skip = self.invalid_loss_strategy == InvalidLossStrategy.NEVER_SKIP
+
+            if is_ddp and never_skip and automatic_optimization:
```

Comment:
call `check_invalid_loss` here instead of in `backward_with_possible_nan_loss`?

then we can rename it to `backward_with_nan_loss`